### PR TITLE
Update container recipes for JEDI

### PR DESCRIPTION
**Pull request by climbfuji**
Update container recipes:
- add `bc` (basic calculator) to OS packages
- add `shumlib` to list of packages to install when following `config/containers/README.md`
- replicate the environment settings from nonroot user to root
- add `checksum: false` to all three container configs
- set target to generic `x86-64` for gnu/clang, and to `core2` for Intel to avoid illegal instructions running the spack-stack containers on older systems / systems with non-Intel CPUs

New Docker test containers were uploaded to JCSDA AWS ECR, singularity containers to JCSDA S3 privatecontainers for testing.

Tested successfully with soca-science bundle for gcc-openmpi container by @climbfuji.

Fixes https://github.com/JCSDA-internal/containers (to be confirmed by @mer-a-o)

### CODE REVIEW

**Review by srherbener — APPROVED** [review]
Changes look good. I did not test anything and I think the CI testing will cover that okay.

**Comment by climbfuji** [comment]
CI tests all passed :-) I will wait for @mer-a-o's testing of these containers before I merged this PR (next week).

**Pull request merged**
